### ISSUE TITLE
Introduce a specific error to replace Timeout::Error

### ISSUE THREAD

**Issue by dsbonev** [jondot/sneakers#259]
While `Timeout::Error` speaks for itself, it can be ambiguous as well. Being part of the standard Ruby library it is available for various gems and for the Ruby library itself. That and the fact sneakers jobs are authored with long-running gem/stdlib/ruby code to do some work it can easily become ambiguous in figuring out which part of the code raised the error - the sneakers gem or the long-running code inside the job which in my case also has a `Timeout.timeout` piece of code. 

Additional benefit of this change is that it will improve error reporting/searching based on class name, seeing 100 occurrences of `Timeout::Error` is less valuable than 50 Sneakers-specific error and 50 `Timeout::Error`.

There is an expectation, which may not be shared by everyone, that when a library would like to raise an error it should be prefixed with library's namespace to be explicit in its meaning, purpose. Looking it from the opposite angle, seeing a library raising `StandardError`, `RuntimeError` (`Timeout::Error`'s parent) doesn't tell me much which means taking a closer look at error messages, stacktraces and such. I have took a look at most popular gems (bundler, rake, rack) and they all define some prefixed errors. This article has some good points on the topic http://weblog.jamisbuck.org/2007/3/7/raising-the-right-exception.html

To summarize, seeing a `Sneakers::TimeoutError/Sneakers::Worker::TimeoutError` will be of much more help than the common `Timeout::Error`.
Thank you for this gem!

**Comment by gabrieljoelc** [comment]
This seems reasonable to me. [Go for it](https://github.com/jondot/sneakers#contributing)! 😄
